browser/ai_chat/ai_chat_service_factory.cc

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ai_chat/ai_chat_service_factory.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/no_destructor.h"
+#include "brave/browser/brave_browser_process.h"
+#include "brave/browser/misc_metrics/process_misc_metrics.h"
+#include "brave/browser/skus/skus_service_factory.h"
+#include "brave/components/ai_chat/content/browser/model_service_factory.h"
+#include "brave/components/ai_chat/core/browser/ai_chat_credential_manager.h"
+#include "brave/components/ai_chat/core/browser/ai_chat_service.h"
+#include "brave/components/ai_chat/core/common/features.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/common/channel_info.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+#include "components/user_prefs/user_prefs.h"
+#include "components/version_info/channel.h"
+#include "content/public/browser/storage_partition.h"
+
+namespace ai_chat {
+
+// static
+AIChatServiceFactory* AIChatServiceFactory::GetInstance() {
+  static base::NoDestructor<AIChatServiceFactory> instance;
+  return instance.get();
+}
+
+// static
+AIChatService* AIChatServiceFactory::GetForBrowserContext(
+    content::BrowserContext* context) {
+  CHECK(context);
+  return static_cast<AIChatService*>(
+      GetInstance()->GetServiceForBrowserContext(context, true));
+}
+
+AIChatServiceFactory::AIChatServiceFactory()
+    : BrowserContextKeyedServiceFactory(
+          "AIChatService",
+          BrowserContextDependencyManager::GetInstance()) {
+  DependsOn(skus::SkusServiceFactory::GetInstance());
+  DependsOn(ModelServiceFactory::GetInstance());
+}
+
+AIChatServiceFactory::~AIChatServiceFactory() = default;
+
+content::BrowserContext* AIChatServiceFactory::GetBrowserContextToUse(
+    content::BrowserContext* context) const {
+  if (!Profile::FromBrowserContext(context)->IsRegularProfile() ||
+      !features::IsAIChatEnabled() ||
+      !ModelServiceFactory::GetForBrowserContext(context) ||
+      !skus::SkusServiceFactory::GetForContext(context)) {
+    return nullptr;
+  }
+  return context;
+}
+
+std::unique_ptr<KeyedService>
+AIChatServiceFactory::BuildServiceInstanceForBrowserContext(
+    content::BrowserContext* context) const {
+  auto skus_service_getter = base::BindRepeating(
```

Comment:
whys is this passing a getter instead of the skus service itself?

Reply:
This code is just moved. We should fix that, but not here.

Reply:
can we create follow-up issues to track this stuff?